aten/src/ATen/native/Integration.cpp

```diff
@@ -38,35 +40,53 @@ Tensor zeros_like_except(const Tensor& y, int64_t dim) {
 
 }
 
-Tensor trapz(const Tensor& y, const Tensor& x, int64_t dim) {
+Tensor trapezoid(const Tensor& y, const Tensor& x, int64_t dim) {
     dim = maybe_wrap_dim(dim, y);
     // asking for the integral with zero samples is a bit nonsensical,
     // but we'll return "0" to match numpy behavior.
     if (y.size(dim) == 0) {
         return zeros_like_except(y, dim);
     }
+    TORCH_CHECK(y.scalar_type() != kBool && x.scalar_type() != kBool, "trapezoid: received a bool input for `x` or `y`, but bool is not supported")
     Tensor x_viewed;
     if (x.dim() == 1) {
-        TORCH_CHECK(x.size(0) == y.size(dim), "trapz: There must be one `x` value for each sample point");
+        // This step takes 'x' with dimension (n,), and returns 'x_view' with
+        // dimension (1,1,...,n,...,1,1) based on dim and y.dim() so that 'x'
+        // can be broadcasted later to match 'y'.
+        // Note: This behavior differs from numpy in that numpy tries to
+        // broadcast 'dx', but this tries to broadcast 'x' to match 'y' instead.
+        TORCH_CHECK(x.size(0) == y.size(dim), "trapezoid: There must be one `x` value for each sample point");
         DimVector sizes(y.dim(), 1);
         sizes[dim] = x.size(0);
         x_viewed = x.view(sizes);
     } else {
         x_viewed = x;
     }
+    // Note the .slice operation reduces the dimension along 'dim' by 1.
+    // The sizes of other dimensions are untouched.
     Tensor x_left = x_viewed.slice(dim, 0, -1);
     Tensor x_right = x_viewed.slice(dim, 1);
 
     Tensor dx = x_right - x_left;
-    return do_trapz(y, dx, dim);
+    return do_trapezoid(y, dx, dim);
 }
 
-Tensor trapz(const Tensor& y, double dx, int64_t dim) {
+Tensor trapezoid(const Tensor& y, const Scalar& dx, int64_t dim) {
     // see above
     if (y.size(dim) == 0) {
         return zeros_like_except(y, dim);
     }
-    return do_trapz(y, dx, dim);
+    TORCH_CHECK(y.scalar_type() != kBool, "trapezoid: received a bool input for `y`, but bool is not supported")
+    TORCH_CHECK(!(dx.isComplex() or dx.isBoolean()), "trapezoid: Currently, we only support dx as a real number.");
```

Comment:
This is not a valid C++ is it?

Reply:
You are right. Thanks for spotting and fixing it.